app/models/test.rb

```diff
@@ -1,7 +1,13 @@
 class Test < ApplicationRecord
-  def self.sort_by_category(category)
-    joins('JOIN categories ON tests.category_id = categories.id')
-      .where(categories: { title: category })
+  belongs_to :user, foreign_key: "author", optional: true
```

Comment:
```suggestion
  belongs_to :`author`, class_name: "User", foreign_key: "`author_id`", optional: true
```
лучше ассоциацию и внешний ключ называть одинаково (`author` и `author_id`)
это удобней в использовании и это соответсвует дефолтному поведению рельс